assets/src/edit-story/components/dropTargets/provider.js

```diff
@@ -174,26 +176,45 @@ function DropTargetsProvider({ children }) {
           resource,
         });
       }
-      combineElements(combineArgs);
-
-      // Reset styles on visisble elements
-      elements
-        .filter(({ id }) => !(id in Object.keys(dropTargets)) && id !== selfId)
-        .forEach((el) => {
-          pushTransform(el.id, {
-            dropTargets: {
-              active: false,
-              replacement: null,
+      const finalizeDrop = (baseColor = null) => {
+        if (baseColor) {
+          combineArgs.firstElement = {
+            ...combineArgs.firstElement,
+            resource: {
+              ...combineArgs.firstElement.resource,
+              baseColor,
             },
+          };
+        }
+        combineElements(combineArgs);
+
+        // Reset styles on visible elements
+        elements
+          .filter(
+            ({ id }) => !(id in Object.keys(dropTargets)) && id !== selfId
```

Comment:
Wait, that's not how [`in`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/in) works.

```js
console.log(1 in [1]); // false
```

Either use `id in dropTargets`, `dropTargets[id]`, or `Object.keys(dropTarget).includes(id)`.

This appears to be a weird hybrid, that doesn't make any sense!

Reply:
Hmm, I actually didn't look at this part at all, left it as it was before, will check...

Reply:
Ok, looks like since this condition has always been set to `true` so far then it has always reset the style for all elements, including irrelevant elements. Changed in a064519.